src/public.ts

```diff
@@ -153,3 +154,44 @@ export function isReleaseCandidate(version: string) {
 export function isSnapshot(version: string) {
     return sls.isSnapshot(sls.parse(version));
 }
+
+/**
+ * Determine if a version matcher string is a valid SLS version matcher
+ *
+ * @param versionMatcher Version matcher string to check
+ */
+export function isValidMatcher(versionMatcher: string) {
```

Comment:
instead of providing these methods, could we slap a `safeValueOf` factory method onto SlsVersionMatcher. It will return `SlsVersionMatcher | null` depending on whether it is valid or not